lib/providers/providers.dart

```diff
@@ -126,4 +129,19 @@ final providers = [
     create: (_) => CreateAccountBloc(),
     update: (_, accountGeneratorService, createAccountBloc) => createAccountBloc..update(accountGeneratorService),
   ),
+  Provider(
+    create: (_) => PermissionService(),
+  ),
+  ProxyProvider<LinksService, ScannerService>(
+    create: (_) => ScannerService(),
+    update: (_, linksService, scannerService) => scannerService..update(linksService),
+  ),
+  ProxyProvider<LinksService, ScannerService>(
+    create: (_) => ScannerService(),
+    update: (_, linksService, scannerService) => scannerService..update(linksService),
+  ),
+  ProxyProvider2<ScannerService, PermissionService, ScannerBloc>(
+    create: (_) => ScannerBloc(),
```

Comment:
ProxyProvider2 is not a very good name - what does this do, can we find a better name?

Reply:
That's generic functions accepting 2 dependencies, in difference with ProxyProvider that accepts 1 depedency, there are also ProxyProvider3 etc...

Reply:
ok